components/dashboard/src/service/json-rpc-workspace-client.ts

```diff
@@ -22,4 +29,48 @@ export class JsonRpcWorkspaceClient implements PromiseClient<typeof WorkspaceSer
         result.item = workspace;
         return result;
     }
+
+    async *watchWorkspaceStatus(
+        request: PartialMessage<WatchWorkspaceStatusRequest>,
+        options?: CallOptions,
+    ): AsyncIterable<WatchWorkspaceStatusResponse> {
+        if (!options?.signal) {
+            throw new ConnectError("signal is required", Code.InvalidArgument);
+        }
+        const it = generateAsyncGenerator<WorkspaceInstance>(
+            (sink) => {
+                const dispose = getGitpodService().registerClient({
+                    onInstanceUpdate: (instance) => {
+                        sink.next(instance);
+                    },
+                });
+                return dispose.dispose;
```

Comment:
does it really work? we return function without bound this? using arrow function would be safer